packages/ripple-binary-codec/src/index.ts

```diff
@@ -56,11 +71,16 @@ function encodeForSigning(json: object): string {
  *
  * @param json JSON object representing the transaction
  * @param signer string representing the account to sign the transaction with
+ * @param definitions Custom rippled types to use instead of the default. Used for sidechains and amendments.
  * @returns a hex string of the encoded transaction
  */
-function encodeForSigningClaim(json: object): string {
+function encodeForSigningClaim(
+  json: object,
+  definitions?: XrplDefinitionsBase,
+): string {
   assert.ok(typeof json === 'object')
-  return signingClaimData(json as ClaimObject)
+  const definitionsOpt = definitions ? { definitions } : undefined
+  return signingClaimData(json as ClaimObject, definitionsOpt)
```

Comment:
Does it not work like this?
```suggestion
  return signingClaimData(json as ClaimObject, { definitions })
```

Reply:
No because then it will always be an object just sometimes with no properties.

Reply:
Yeah, that should also be acceptable by the code.

Reply:
As written defining Definitions is non-optional, but defining your own _is_ optional. In order to allow `{ undefined }` I'd have to set the default value inside the function instead of as part of the parameter, which I think is worse for readability since it requires adding an extra check and setting the default of `DEFAULT_DEFINITIONS` twice. 

Current signingClaimData implementation for reference: 
```
function signingClaimData(
  claim: ClaimObject,
  opts: { definitions: XrplDefinitionsBase } = {
    definitions: DEFAULT_DEFINITIONS,
  },
): Buffer {
  const num = bigInt(String(claim.amount))
  const prefix = HashPrefix.paymentChannelClaim
  const types = opts.definitions.getAssociatedTypes()
  const channel = types.Hash256.from(claim.channel).toBytes()
  const amount = types.UInt64.from(num).toBytes()

  const bytesList = new BytesList()

  bytesList.put(prefix)
  bytesList.put(channel)
  bytesList.put(amount)
  return bytesList.toBytes()
}
```

Additional logic to allow `{ undefined }`:
```
function signingClaimData(
  claim: ClaimObject,
  opts: { definitions?: XrplDefinitionsBase } = {
    definitions: DEFAULT_DEFINITIONS,
  },
): Buffer {
let types = undefined
if(opts.definitions === undefined) {
  types = DEFAULT_DEFINITIONS.getAssociatedTypes()
} else {
  types = opts.definitions.getAssociatedTypes()
}
...
}
```